src/components/Header/index.js

```diff
@@ -10,10 +11,75 @@ import {ReactComponent as MoreIcon} from 'assets/images/icons/more.svg'
 import {ReactComponent as LogInIcon} from 'assets/images/icons/logIn.svg'
 import {ReactComponent as RegisterIcon} from 'assets/images/icons/register.svg'
 import HeaderDropdown from 'components/Dropdown/HeaderDropdown'
+import * as productActions from 'redux/actions/productActions'
 
 import './styles.scss'
 
 const Header = () => {
+  const dispatch = useDispatch()
+  const navigate = useNavigate()
+
+  const [value, setValue] = useState('')
+  const [showDropDown, setShowDropDown] = useState(false)
+
+  const productFilterByName = useSelector(
+    state => state.Product.productFilterByName.data,
+  )
```

Comment:
just here is enough, a problem: a same variable name but it presents for 2 type of var: one is object and one is array
 products = useSelector( state => state.Product.productFilterByName.data)